process/subprocesses/_14_aggregation.py

```diff
@@ -1,358 +1,353 @@
+                    ELSE
+                        (SUM(s."{weight}"*s."{i}"::numeric)/SUM(s."{weight}"))::numeric
+                END) AS "{weight}_{i}"
+                '''
+            agg_formula = ','.join(
+                [weighting.format(i=i, weight=weight) for i in indicator_list],
+            )
+        else:
+            agg_formula = ','.join(
+                [
+                    f'''AVG(s."{i}"::numeric) AS "avg_{i}"'''
+                    for i in indicator_list
+                ],
+            )
+        queries = [
+            f"""DROP TABLE IF EXISTS {table};""",
+            f"""CREATE TABLE "{table}" AS
+            SELECT b.{id},
+            {keep_columns}
+            ST_Area(b.geom)/10^6 AS area_sqkm,
+            COUNT(s.*) AS {count_units},
+            {agg_formula},
+            b.geom
+            FROM "{boundaries}" b
+            LEFT JOIN "{agg_source}" s ON {agg_on}
+            {query}
+            GROUP BY b.{id}, {keep_columns} b.geom;""",
+            f"""DELETE FROM {table} WHERE {count_units} = 0;""",
+            f"""CREATE INDEX {table}_ix  ON {table} ({id});""",
+            f"""CREATE INDEX {table}_gix ON {table} USING GIST(geom);""",
+        ]
+        for query in queries:
+            try:
+                print(query)
+                with r.engine.begin() as connection:
+                    connection.execute(text(query))
+                processed_aggs.append(agg)
+            except Exception as e:
+                sys.exit(
+                    f"Error when attempting to aggregate for {agg} '{boundary_data}' (check custom aggregation configuration): {e}",
+                )
+
+
+def aggregate_study_region_indicators(codename):
+    start = time.time()
+    script = '_13_aggregation'
+    task = 'Compile study region destinations'
+    r = ghsci.Region(codename)
+    print('\nCalculating small area neighbourhood grid indicators... ')
+    # calculate within-city indicators weighted by sample points for each city
+    # calc_grid_pct_sp_indicators take sample point stats within each city as
+    # input and aggregate up to grid cell indicators by calculating the mean of
+    # sample points stats within each hex
+    calc_grid_pct_sp_indicators(r, ghsci.indicators)
+
+    print('\nCalculating city summary indicators... ')
+    # Calculate city-level indicators weighted by population
+    # calc_cities_pop_pct_indicators function take grid cell indicators and
+    # pop estimates of each city as input then aggregate grid cell to city-level
+    # indicator by summing all the population weighted grid cell indicators
+
+    # in addition to the population weighted averages, unweighted averages are
+    # also included to reflect the spatial distribution of key walkability
+    # measures (regardless of population distribution)
+    calc_cities_pop_pct_indicators(r, ghsci.indicators)
+
+    print('\nCalculating custom aggregation indicators... ')
+    custom_aggregation(r, ghsci.indicators)
```

Comment:
Does this still work if the 'custom_aggregation' section isn't defined in the configuration file?  I get that the check for this could be put in the custom_aggregation fuctino itself... but just looking at the function, it seems to assume that there is an entry r.config['custom_aggregations'] ---- and if we haven't handled this to insert it (perhaps as an empty list), then I think this will fail without the check that used to be here?